### ISSUE TITLE
CFE: Consider suppressing spurious nullability errors

### ISSUE THREAD

**Issue by leafpetersen** [dart-lang/sdk#42607]
If an opted out library is compiled in strong null safety mode, the CFE correctly emits an error indicating that this is not allowed.  However it also emits some nullability errors from the opted out code.  This can be somewhat confusing to users (see [here](https://github.com/dart-lang/test/issues/1291)).  If possible, it might be good to suppress other compile errors when the "opted out in sound mode" error is hit.  Repro:

Compile with:
```
~/src/dart-repo/sdk/xcodebuild/ReleaseX64/dart-sdk/bin/dart --enable-experiment=non-nullable --null-safety ~/tmp/test.dart
```

```dart
// @dart = 2.6

main() {
  int x = 3;
  x?.isEven;
}
```

Output is:

```
leafp-macbookpro:sdk leafp$ ~/src/dart-repo/sdk/xcodebuild/ReleaseX64/dart-sdk/bin/dart --enable-experiment=non-nullable --null-safety ~/tmp/test.dart
../../../tmp/test.dart:5:3: Warning: Operand of null-aware operation '?.' has type 'int' which excludes null.
  x?.isEven;
  ^
../../../tmp/test.dart:1:1: Error: A library can't opt out of non-nullable by default, when in nnbd-strong mode.
// @dart = 2.6
^^^^^^^^^^^^^^
```

cc @jakemac53

**Comment by johnniwinther** [comment]
Duplicate of https://github.com/dart-lang/sdk/issues/43327

**Comment by johnniwinther** [comment]
No, not a duplicate, just the reverse problem.

**Comment by mit-mit** [comment]
@leafpetersen @johnniwinther are we sure this isn't for beta? We're going to have many more mixed mode apps early in the release cycle than later on...

**Comment by johnniwinther** [comment]
Good point. I'll look into whether this can be part of beta.

**Comment by johnniwinther** [comment]
Looking closer at the spurious warning it turns out that the issue was already fixed by the fix to #43327 

Adding a test for this issue in https://dart-review.googlesource.com/c/sdk/+/169740 to verify.
